bundles/org.palladiosimulator.simulizar.action/model/actions.ecore

```diff
@@ -18,66 +18,74 @@
         </eAnnotations>
+        <eParameters name="resourceTableManager" eType="#//core/IResourceTableManager"/>
         <eParameters name="affectedRoleSet" eType="#//instance/RoleSet"/>
         <eParameters name="controllerCallsVariableUsages" lowerBound="1" eType="#//parameter/ControllerCallInputVariableUsageCollection"/>
         <eParameters name="asyncExecutionContext" eType="#//context/ExecutionContext"/>
       </eOperations>
       <eOperations name="executeAsync" lowerBound="1" eType="#//context/ExecutionContext">
         <eAnnotations source="http://www.eclipse.org/emf/2002/GenModel">
-          <details key="body" value="return org.palladiosimulator.simulizar.action.interpreter.ActionRuntimeState.getInterpreterBuilder(affectedRoleSet, getRepository()).isAsync().build().doSwitch(this).getContext().get();"/>
+          <details key="body" value="return org.palladiosimulator.simulizar.action.interpreter.ActionRuntimeState.getInterpreterBuilder(affectedRoleSet, getRepository()).isAsync().build(resourceTableManager).doSwitch(this).getContext().get();"/>
         </eAnnotations>
         <eParameters name="affectedRoleSet" lowerBound="1" eType="#//instance/RoleSet"/>
+        <eParameters name="resourceTableManager" eType="#//core/IResourceTableManager"/>
       </eOperations>
       <eOperations name="executeAsync" eType="#//context/ExecutionContext">
         <eAnnotations source="http://www.eclipse.org/emf/2002/GenModel">
-          <details key="body" value="org.palladiosimulator.simulizar.action.interpreter.ActionRuntimeState.getInterpreterBuilder(affectedRoleSet, getRepository()).isAsync(asyncExecutionContext).build().doSwitch(this);&#xD;&#xA;return asyncExecutionContext;&#xD;&#xA;"/>
+          <details key="body" value="org.palladiosimulator.simulizar.action.interpreter.ActionRuntimeState.getInterpreterBuilder(affectedRoleSet, getRepository()).isAsync(asyncExecutionContext).build(resourceTableManager).doSwitch(this);&#xA;return asyncExecutionContext;&#xA;"/>
           <details key="documentation" value="Triggers the asynchronous execution of this AdaptationBehavior for the given RoleSet in the given ExecutionContext.&#xD;&#xA;To conform to ome overloaded versions of this method, this context  is also returned.&#xD;&#xA;It should be passed to synchronous adaptation behavior executions that are done within."/>
         </eAnnotations>
         <eParameters name="affectedRoleSet" eType="#//instance/RoleSet"/>
         <eParameters name="asyncExecutionContext" eType="#//context/ExecutionContext"/>
+        <eParameters name="resourceTableManager" eType="#//core/IResourceTableManager"/>

```

Comment:
This is really problematic, as all action specifications in qvto will break. I would strongly suggest resolving the manager somewhere in the execution context. Furthermore, the parameter does not make sense to be carried into the QVTo world.

Reply:
The current solution does not break the qvto scripts; by passing the resourceTableManager as 'configuration property resourceTableManager : IResourceTableManager;' to the respective qvto scripts (only script ' Palladio-Analyzer-SimuLizar\bundles\ExampleModels\TransientEffects\org.palladiosimulator.simulizar.examples.mediastore\reconfigurations_active\horizontalscaling.qvto' is affected by this change), the scripts are still executable; for sure, passing the parameter to the script is just a temporarily workaround to guarantee, that the scripts are still executable. Thus I created the follow-up ticket https://jira.palladio-simulator.com/browse/SIMULIZAR-130 to address issue.